src/VirtualClient/VirtualClient.Actions/SPECview/SpecViewExecutor.cs

```diff
@@ -49,38 +58,72 @@ public SpecViewExecutor(IServiceCollection dependencies, IDictionary<string, ICo
         }
 
         /// <summary>
-        /// The command line argument defined in the profile.
+        /// The viewset that will be run by SPECviewperf.
         /// </summary>
-        public string GUIOption
+        public string Viewset
         {
             get
             {
-                return this.Parameters.GetValue<string>(nameof(SpecViewExecutor.GUIOption));
+                // Remove whitespaces in the argument (e.g. "3dsmax, catia" -> "3dsmax,catia")
+                return this.Parameters.GetValue<string>(nameof(SpecViewExecutor.Viewset)).Replace(" ", string.Empty);
             }
         }
 
         /// <summary>
-        /// The viewset that will be run by SPECviewperf.
+        /// Using PsExec to run specviewperf in session 1.
         /// </summary>
-        public string Viewset
+        public string PsExecPackageName
         {
             get
             {
-                // Remove whitespaces in the argument (e.g. "3dsmax, catia" -> "3dsmax,catia")
-                return this.Parameters.GetValue<string>(nameof(SpecViewExecutor.Viewset)).Replace(" ", string.Empty);
+                return this.Parameters.GetValue<string>(nameof(SpecViewExecutor.PsExecPackageName));
+            }
+        }
+
+        /// <summary>
+        /// PsExec session number
+        /// </summary>
+        public int PsExecSession
+        {
+            get
+            {
+                return this.Parameters.GetValue<int>(nameof(SpecViewExecutor.PsExecSession));
             }
         }
 
         /// <summary>
         /// The path to the RunViewperf.exe.
         /// </summary>
-        public string ExecutablePath { get; set; }
+        protected string SpecviewExecutablePath { get; set; }
 
         /// <summary>
         /// Defines the path to the SPECview package that contains the workload
         /// executable.
         /// </summary>
-        protected DependencyPath Package { get; set; }
+        protected DependencyPath SpecviewPackage { get; set; }
+
+        /// <summary>
+        /// The path to the PsExec.exe.
+        /// </summary>
+        protected string PsExecExecutablePath { get; set; }
+
+        /// <summary>
+        /// Defines the path to the PsExec package that contains the workload
+        /// executable.
+        /// </summary>
+        protected DependencyPath PsExecPackage { get; set; }
+
+        /// <summary>
+        /// Defines the viewset to log file mapping.
+        /// executable.
+        /// </summary>
+        protected IDictionary<string, string> ViewsetLogFileNameMapping
```

Comment:
Is there a reason you need it protected? is it for the unit test?

Reply:
Yes, need to access this mapping in unit tests. E.g: [https://github.com/microsoft/VirtualClient/blob/f4cc2cac55b1ae456950898c1d86ea490a9e7f7b/src/VirtualClient/VirtualClient.Actions.UnitTests/SPEC/SpecViewExecutorTests.cs#L141C26-L141C26]